Source/JavaScriptCore/wasm/WasmIPIntGenerator.cpp

```diff
@@ -1216,6 +1324,9 @@ PartialResult WARN_UNUSED_RETURN IPIntGenerator::addEndToUnreachable(ControlEntr
         entry.enclosedExpressionStack.constructAndAppend(signature.returnType(i), Value { });
     auto block = entry.controlData;
 
+    if (ControlType::isTry(block) || ControlType::isAnyCatch(block))
+        --m_tryDepth;
```

Comment:
Is this right?

Reply:
Like, why does a catch increase this depth? If we throw or delegate to an upper level, it just counts trys right?

Reply:
Hmm, no, I am wrong. We delegate to the label of the try, which should jump to the associated catch. So the catch and try need to be at the same depth.